kernel/userspace.c

```diff
@@ -901,19 +901,19 @@ SYS_INIT(app_shmem_bss_zero, POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
  * Default handlers if otherwise unimplemented
  */
 
-static uintptr_t handler_bad_syscall(uintptr_t bad_id, uintptr_t arg2,
-				     uintptr_t arg3, uintptr_t arg4,
-				     uintptr_t arg5, uintptr_t arg6,
+static uintptr_t handler_bad_syscall(uintptr_t bad_id, uintptr_t arg1,
+				     uintptr_t arg2, uintptr_t arg3,
+				     uintptr_t arg4, uintptr_t arg5,
 				     void *ssf)
 {
 	LOG_ERR("Bad system call id %" PRIuPTR " invoked", bad_id);
 	arch_syscall_oops(ssf);
 	CODE_UNREACHABLE; /* LCOV_EXCL_LINE */
 }
 
-static uintptr_t handler_no_syscall(uintptr_t arg1, uintptr_t arg2,
-				    uintptr_t arg3, uintptr_t arg4,
-				    uintptr_t arg5, uintptr_t arg6, void *ssf)
+static uintptr_t handler_no_syscall(uintptr_t arg0, uintptr_t arg1,
+				    uintptr_t arg2, uintptr_t arg3,
+				    uintptr_t arg4, uintptr_t arg5, void *ssf)
```

Comment:
I am guessing that the intent behind this change is to make the numbering of the arguments more consistent with that of  the other sycalls. However, like handler_bad_syscall(), I think scripts/gen_syscalls.py should be updated if moving ahead with this change.

Reply:
This is going to cause confusion as `arch_syscall_invoke*()` start with `arg1`.